spark/src/test/scala/org/apache/spark/sql/delta/FileMetadataMaterializationTrackerSuite.scala

```diff
@@ -35,27 +39,29 @@ class FileMetadataMaterializationTrackerSuite extends SparkFunSuite with TimeLim
     }
 
     // Initialize the semaphore for tests
-    val semaphore = new Semaphore(10)
+    val totalAvailablePermits = spark.sessionState.conf.getConf(
+      DeltaSQLConf.DELTA_COMMAND_FILE_MATERIALIZATION_LIMIT)
+    val semaphore = new Semaphore(totalAvailablePermits)
```

Comment:
Wait, that means test suites aren't being run in isolation?
I'm actually surprised we don't have a lot more issues with our tests if that's the case

Reply:
Yes, that's the current state of our Github Workflow Actions is, we just brute-force all the tests sequentially for a thread of execution.